src/tarantool-gdb.py

```diff
@@ -0,0 +1,1371 @@
+        ext = cls.decode_compression(data, len)
+        if ext is None:
+            return
+        return str(ext)
+
+    @classmethod
+    def decode_interval_field(cls, data):
+        mp_type = cls.typeof(data)
+        if mp_type == cls.MP_UINT:
+            value = cls.decode_uint(data)
+        elif mp_type == cls.MP_INT:
+            value = cls.decode_int(data)
+        else:
+            raise gdb.GdbError('Interval.decode_field: unexpected type ({})'.format(mp_type))
+        return value
+
+    INTERVAL_FIELD_YEAR = find_value('FIELD_YEAR')
+    INTERVAL_FIELD_MONTH = find_value('FIELD_MONTH')
+    INTERVAL_FIELD_WEEK = find_value('FIELD_WEEK')
+    INTERVAL_FIELD_DAY = find_value('FIELD_DAY')
+    INTERVAL_FIELD_HOUR = find_value('FIELD_HOUR')
+    INTERVAL_FIELD_MINUTE = find_value('FIELD_MINUTE')
+    INTERVAL_FIELD_SECOND = find_value('FIELD_SECOND')
+    INTERVAL_FIELD_NANOSECOND = find_value('FIELD_NANOSECOND')
+    INTERVAL_FIELD_ADJUST = find_value('FIELD_ADJUST')
+
+    DT_SNAP = find_value('DT_SNAP')
+
+    @classmethod
+    def decode_interval(cls, data, len): # interval_unpack
+        pos = data.pos()
+        itv = cls()
+        num_fields = data.read_u8()
+        for _ in range(0, num_fields):
+            field = data.read_u8()
+            value = cls.decode_interval_field(data)
+            if field == cls.INTERVAL_FIELD_YEAR:
+                itv.year = value
+            elif field == cls.INTERVAL_FIELD_MONTH:
+                itv.month = value
+            elif field == cls.INTERVAL_FIELD_WEEK:
+                itv.week = value
+            elif field == cls.INTERVAL_FIELD_DAY:
+                itv.day = value
+            elif field == cls.INTERVAL_FIELD_HOUR:
+                itv.hour = value
+            elif field == cls.INTERVAL_FIELD_MINUTE:
+                itv.min = value
+            elif field == cls.INTERVAL_FIELD_SECOND:
+                itv.sec = value
+            elif field == cls.INTERVAL_FIELD_NANOSECOND:
+                itv.nsec = value
+            elif field == cls.INTERVAL_FIELD_ADJUST:
+                if value > cls.DT_SNAP:
+                    return None, 'unexpected adjust value ({})'.format(value)
+                itv.adjust = value
+            else:
+                return None, 'unexpected field type ({})'.format(field)
+        if data.pos() - pos != len:
+            data.seek(pos + len)
+            return None, 'unexpected length: expected - {}, actual - {}'.format(len, data.pos() - pos)
+        return itv, None
+
+    @classmethod
+    def to_string_interval(cls, data, len): # mp_snprint_interval
+        itv, err = cls.decode_interval(data, len)
+        if err is not None:
+            gdb.write(err, gdb.STDERR)
+            return
+        return str(itv)
+
+    @classmethod
+    def to_string_ext(cls, data, depth, ext_len, ext_type): # msgpack_snprint_ext
+        pos = data.pos()
+
+        if ext_type == cls.MP_DECIMAL:
```

Comment:
It's a replication of c-function `msgpack_snprint_ext` and mapping switch operator into if-elif-else is used because it's simple and straightforward and visually closer to original switch, so I'd like to keep it to simplify synchronization if needed. Besides, it doesn't seem that we need performance optimization here.